### PR TITLE
Coverage code changes (refactored the code to work with new main branch code)

### PR DESCRIPTION
**Pull request by sumedhak27**
## Problem Statement
<pre>
  <code>
    Story Ref (if any):
    Please add Problem statement here...
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Please add Problem decription here...
  </code>
</pre>
## Solution
<pre>
  <code>
    Please add summary of the change,List any dependencies that are required for this change.
  </code>
</pre>
## Sanity testing on RPM done
<pre>
  <code>
    - [ ] Yes
    - [ ] No
  </code>
</pre>
## Unit/Manual Testing Description
<pre>
  <code>
    Please describe the tests that you ran to verify your changes.
  </code>
</pre>

### CODE REVIEW

File: low-level/framework/sspl_ll_d [hunk 1]
```diff
@@ -740,6 +738,26 @@ def refresh_iem_log_file(signal_number, frame):
...
+    # /opt/seagate/cortx/sspl/sspl_test/start_sspl_coverge.sh
+    logger.info("Ending Coverage Scope")
+    co.stop()
+    logger.info("coverage object stopped.")
+    co.save()
+    logger.info("coverage info saved.")
+    cov_per = co.xml_report(
+ 						outfile='/tmp/sspl/sspl_xml_coverage_report.xml',
+ 						ignore_errors=True,
+ 					)
```
Comment (thavanathan) [inline low-level/framework/sspl_ll_d:752]: Please format these lines.

File: low-level/framework/sspl_ll_d [hunk 2]
```diff
@@ -740,6 +738,26 @@ def refresh_iem_log_file(signal_number, frame):
     else:
         logger.info(f"IEMSensor is not in running, ignoring signal to refresh file")
 
+def generate_cov_report(signal_number, frame):
+    # co object will be available on execution of
+    # /opt/seagate/cortx/sspl/sspl_test/start_sspl_coverge.sh
+    logger.info("Ending Coverage Scope")
+    co.stop()
```
Comment (thavanathan) [inline low-level/framework/sspl_ll_d:745]: Is it a global variable that coverage.py sets?

File: sspl_test/coverage/coverage_code [hunk 3]
```diff
@@ -6,20 +6,4 @@ co = Coverage(
...
- 						ignore_errors=True,
- 					)
-    logger.info(f"XML coverage report generated with coverage of {cov_per}%")
- 	## Enable below code to inable HTML report generation
- 	# html_cov_per = co.html_report(
- 	#                     directory='/tmp/sspl/sspl_html_coverage',
- 	#                     ignore_errors=True,
- 	#                 )
- 	# logger.info(f"HTML coverage report geverated with coverage of {html_cov_per}%")
+        co.start()
```
Comment (thavanathan) [inline sspl_test/coverage/coverage_code:9]: Please add a new line at the end.

Comment (thavanathan) [inline sspl_test/coverage/coverage_code:9]: `coverage_code` seems meaningless. Can we rename this something like `sspl_code_coverage.py`?

**Review by thavanathan — CHANGES_REQUESTED** [review]

File: low-level/framework/sspl_ll_d (hunk 2 above)
Comment (sumedhak27) [reply low-level/framework/sspl_ll_d:745]: yes, it sets a global variable.

File: sspl_test/coverage/coverage_code (hunk 3 above)
Comment (sumedhak27) [reply sspl_test/coverage/coverage_code:9]: but it is not a python file, it just has the patches of code to be inserted in the sspl_ll_d file.

File: low-level/framework/sspl_ll_d (hunk 1 above)
Comment (sumedhak27) [reply low-level/framework/sspl_ll_d]: done

File: sspl_test/coverage/coverage_code (hunk 3 above)
Comment (sumedhak27) [reply sspl_test/coverage/coverage_code]: done

**Review by sandeepsgt — APPROVED** [review]

File: low-level/framework/sspl_ll_d [hunk 4]
```diff
@@ -740,6 +738,24 @@ def refresh_iem_log_file(signal_number, frame):
     else:
         logger.info(f"IEMSensor is not in running, ignoring signal to refresh file")
 
+def generate_cov_report(signal_number, frame):
```
Comment (cdeshmukh) [inline low-level/framework/sspl_ll_d:741]: why we are adding code in production version of sspl_ll_d? We maintained earlier same code in different script, so as to enable when needed in dev/test env, same approach we need.
Add this code with comment markers in dev/test version of sspl_ll_d with start/stop_coverag_report.sh

Comment (sumedhak27) [reply low-level/framework/sspl_ll_d:741]: addressed.

File: low-level/framework/sspl_ll_d (hunk 2 above)
Comment (thavanathan) [reply low-level/framework/sspl_ll_d]: Okay

**Review by thavanathan — APPROVED** [review]

File: low-level/framework/sspl_ll_d [hunk 5]
```diff
@@ -740,6 +738,9 @@ def refresh_iem_log_file(signal_number, frame):
...
+    #DO NOT EDIT: Marker comment to dynamically add code to stop coverage, save and generate code coverage report
+    pass
```
Comment (cdeshmukh) [inline low-level/framework/sspl_ll_d:744]: dont keep placeholder function too, just keep marker comments and add function code and signhandler registration code through separate start_code_coverage.sh script only

File: low-level/framework/sspl_ll_d [hunk 6]
```diff
@@ -794,6 +795,8 @@ if __name__ == "__main__":
     signal.signal(signal.SIGUSR1, set_log_level)
     # Handle signal for iem file refresh
     signal.signal(signal.SIGUSR2, refresh_iem_log_file)
+    # Handle signal to stop, save and generate coverage report
+    signal.signal(signal.SIGRTMIN, generate_cov_report)
```
Comment (cdeshmukh) [inline low-level/framework/sspl_ll_d:799]: same comment as above

Comment (cdeshmukh) [reply low-level/framework/sspl_ll_d:799]: Also, any reason why SIGRT* signal is being used? why not make reuse of SIGUSR1 as that will get freed up now, with log level dynamic change notification would come from conf store in future

File: low-level/framework/sspl_ll_d (hunk 5 above)
Comment (sumedhak27) [reply low-level/framework/sspl_ll_d:742]: addressed.

File: low-level/framework/sspl_ll_d (hunk 6 above)
Comment (sumedhak27) [reply low-level/framework/sspl_ll_d]: used SIGUSR1.

**Review by cdeshmukh — APPROVED** [review]

File: low-level/framework/sspl_ll_d [hunk 7]
```diff
@@ -717,21 +717,19 @@ def send_thread_controller_request(state):
...
-    key='sspl/config/SYSTEM_INFORMATION/log_level'
-    logger.info("signal handler is called with {0}".format(signal_number))
-    log_level = store.get(key)
-    if log_level and isinstance(log_level, bytes):
-        log_level = log_level.decode('utf-8')
-    logger.info(f"Received SIGUSR1 signal to set log level to '{log_level}'")
-    logger.setLevel(log_level)
-    #DO NOT EDIT: Marker comment to dynamically add code to stop coverage, save and generate code coverage report
-
+# def set_log_level(signal_number, frame):
```
Comment (cdeshmukh) [inline low-level/framework/sspl_ll_d:720]: Add a note here
\# Dynamic config change including log level set will ne enabled through ConfStor. 
\#On getting a trigger, this same function can be used back to set the required log level, excluding its interaction with Consul.

File: low-level/framework/sspl_ll_d [hunk 8]
```diff
@@ -791,9 +790,10 @@ if __name__ == "__main__":
         sys.exit(os.EX_USAGE)
 
     # Handle signals for log level change
-    signal.signal(signal.SIGUSR1, set_log_level)
+    # signal.signal(signal.SIGUSR1, set_log_level)
     # Handle signal for iem file refresh
     signal.signal(signal.SIGUSR2, refresh_iem_log_file)
+    #DO NOT EDIT: Marker comment to dynamically add signal handler for coverage job
```
Comment (cdeshmukh) [inline low-level/framework/sspl_ll_d:796]: #DO NOT EDIT: Marker comment to dynamically add signal handler for SIGUSR1 to generate code coverage report

**Review by thavanathan — APPROVED** [review]

File: low-level/framework/sspl_ll_d (hunk 7 above)
Comment (sumedhak27) [reply low-level/framework/sspl_ll_d:720]: added.

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Pull request merged**